src/SFA.DAS.EmployerFinance/Commands/RefreshEmployerLevyData/LevyImportCleanupStrategy.cs

```diff
@@ -0,0 +1,191 @@
+        }
+
+        private async Task UpdateEndOfYearAdjustment(string empRef, DasDeclaration yearEndAdjustment, DasDeclaration[] hmrcDeclarations)
+        {
+            if (yearEndAdjustment.LevyDueYtd == null && !yearEndAdjustment.NoPaymentForPeriod)
+            {
+                throw new ArgumentNullException(nameof(yearEndAdjustment));
+            }
+
+            yearEndAdjustment.EndOfYearAdjustment = true;
+
+            if (yearEndAdjustment.NoPaymentForPeriod)
+                return;
+
+            var period12Declaration = await GetDeclarationEffectiveForPeriod12(empRef, yearEndAdjustment.PayrollYear, yearEndAdjustment.SubmissionDate, hmrcDeclarations);
+
+            if (period12Declaration?.LevyDueYtd != null)
+            {
+                // Caution: the calculation here is incorrect (it should be the other way round). This results in negative values in LevyDeclaration EndOfYearAdjustmentAmount.
+                // However, do not fix this as somewhere later (probably in a view or sproc) the EndOfYearAdjustmentAmount is being inverted when writing to the transaction line table.
+                yearEndAdjustment.EndOfYearAdjustmentAmount =
+                    period12Declaration.LevyDueYtd.Value - yearEndAdjustment.LevyDueYtd ?? 0;
+            }
+            else
+            {
+                yearEndAdjustment.EndOfYearAdjustmentAmount = -yearEndAdjustment.LevyDueYtd ?? 0;
+            }
+        }
+
+        /// <summary>
+        ///     Returns the latest declaration for the year (which is the one that will be effective for period 12).
+        /// </summary>
+        /// <param name="yearEndAdjustmentCutOff">
+        ///     If we are processing an adjustment we need to find the P12 value, which might be another adjustment. In that case, it must 
+        ///     be an earlier adjustment (i.e. the immediately preceding adjustment).
+        /// </param>
+        /// <param name="hmrcDeclarations">
+        ///     The declarations retrieved from HMRC (these are in-memory and have not yet been written to the database.
+        /// </param>
+        /// <returns>
+        ///     The declaration that was effective for period 12 or null if one was not found.
+        /// </returns>
+        /// <remarks>
+        ///     The database contains all the declarations retrieved to date from HMRC, <see cref="hmrcDeclarations"/>
+        ///     contains the declarations that HMRC has that we have not retrieved previously (which is determined by submission date).
+        /// </remarks>
+        private async Task<DasDeclaration> GetDeclarationEffectiveForPeriod12(string empRef, string payrollYear, DateTime yearEndAdjustmentCutOff, IEnumerable<DasDeclaration> hmrcDeclarations)
+        {
+            // Look in the declarations that have just been retrieved from HMRC (which will contain everything not yet in the database)
```

Comment:
This comment is unnecessary, the method below is very well named so is descriptive enough.

Reply:
Removed